docs/book/src/cronjob-tutorial/testdata/project/controllers/cronjob_controller_test.go

```diff
@@ -0,0 +1,217 @@
+/*
+We can create the tests in a separate file, cronjob_controller_test.go.
+Keeping consistent with the Gingko usage in our autogenerated suite_test.go code, we’ll use Gingko to write our tests.
+*/
+
+/*
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+// +kubebuilder:docs-gen:collapse=Apache License
+
+/*
+As usual, we start with necessary imports.
+*/
+package controllers
+
+import (
+	"context"
+	"reflect"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	batchv1 "k8s.io/api/batch/v1"
+	batchv1beta1 "k8s.io/api/batch/v1beta1"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/types"
+
+	cronjobv1 "tutorial.kubebuilder.io/project/api/v1"
+)
+
+/*
+We then set up consts we'll use in our tests, including test names for our objects as well as helper definitions
+like SUCCESS and FAILURE to make our tests more readable. The usage of timeout and interval durations will become evident soon.
+*/
+const (
+	CronjobName      string = "test-cronjob"
+	CronjobNamespace string = "test-cronjob-namespace"
+	JobName          string = "test-job"
+
+	SUCCESS bool = true
+	FAILURE bool = false
+
+	timeout  = time.Second * 10
+	interval = time.Millisecond * 250
+)
+
+/*
+We also expose the GVK for our CronJob version, which we will use to manage ownership of Jobs that belong to our test CronJob.
+One way you can set this GVK is by defining it as a global var like so:
+*/
+var (
+	Kind string                  = reflect.TypeOf(cronjobv1.CronJob{}).Name()
+	GVK  schema.GroupVersionKind = cronjobv1.GroupVersion.WithKind(Kind)
+)
+
+/*
+Now we set up the Gingko testing framework to run CronJob controller tests.
+
+The first step to writing a simple integration test is to actually create an instance of CronJob you can run tests against.
+Note that to create a CronJob, you’ll need to create a stub CronJob struct that contains your CronJob’s specifications.
+
+To create a stub CronJob, we'll also need to create stubs of its required downstream objects--namely, its Job template spec and
+its Pod template spec.
+*/
+var _ = Describe("Run CronJob Controller", func() {
+	Context("Can create CronJob", func() {
+		It("Should create all sub resources successfully", func() {
+			By("Create CronJob")
+			ctx := context.Background()
```

Comment:
It shows not accurate for me. I understand that we should here create a CronJob CR and then, check that all resources that will manage by this CR will be created and in the expected state. See: https://github.com/microsoft/azure-databricks-operator/blob/0f722a710fea06b86ecdccd9455336ca712bf775/controllers/dbfsblock_controller_test.go#L76-L95

Reply:
I reformatted this test section to mirror https://github.com/microsoft/azure-databricks-operator/blob/0f722a710fea06b86ecdccd9455336ca712bf775/controllers/dbfsblock_controller_test.go#L76-L95 better--now, we only use one `Context()` and one `It()`, and the check for no active downstream objects immediately follows the creation of the test CronJob.

Reply:
It still a little confusing for me. IMO it should be something as: 


```
	BeforeEach(func() {
		// - Apply a new CronJobCR 
	})
```

**Should work successfully when a new CronJob CR is applied** 

- Check that the CronJob object was created successfully
- Check that the spec a or b is of the CronJob resource is == the value expected.  

**Should increase the spec X when Jobs are created** 
- create the Job
- check that the spec X was increased. 

I hope that  it helps you with

Reply:
I honestly do prefer the preexisting format of the tests, because it saves readers from having to learn about `BeforeEach()` and the scopes it applies to, and also mirrors what we see in https://github.com/microsoft/azure-databricks-operator/blob/master/api/v1alpha1/dbfsblock_types_test.go.

I can add the extra check for `Check that the spec a or b is of the CronJob resource is == the value expected.` but am honestly a bigger fan of how the tests are formatted now and how they match the azure databricks example. What do you think?

Reply:
You can close this one now. IMO, if you address the @DirectXMan12 suggestions it would be enough.